File: leetcode-submissions/1475_Final_Prices_With_a_Special_Discount_in_a_Shop.ts
function finalPrices(prices: number[]): number[] {
    let ans:number[]=new Array(prices.length);
    ans= prices;
    for(let i:number =0;i<prices.length;i++){
        for(let j:number =i+1;j<prices.length;j++){
            if(prices[j]<=prices[i]){
                ans[i]=prices[i]-prices[j];
                break;
            }
        }
        
    }
    return ans ;
};

File: leetcode-submissions/1475_Final_Prices_With_a_Special_Discount_in_a_Shop.cpp
class Solution {
public:
    vector<int> finalPrices(vector<int>& prices) {
        vector<int>ans(prices.size(),-1);
        for(int i=0;i<prices.size();i++){
            for(int j=i+1;j<prices.size();j++){
                if(prices[i]>=prices[j]){
                    ans[i]=(prices[i]-prices[j]);
                    break;
                }
            }
            if(ans[i]==-1){
                ans[i]=prices[i];
            }
        }
        return ans;
    }
};